### PR TITLE
adding rate limit info

### PR DESCRIPTION
**Pull request by admiralAwkbar**
- adding rate limit checks and waits

### CODE REVIEW

File: gh-repo-visibility [hunk 1]
```diff
@@ -576,18 +669,21 @@ if [[ -n "${INPUT_FILE_NAME}" ]]; then
...
+#### Function ParseUpdateArray #################################################
+ParseUpdateArray() {
+  # Check api limit
+  CheckAPILimit
+  # Ser var for API limit used
+  API_LIMIT_REMAINING="${API_REMAINING}"
   # Loop through the list of repos and update the visibility
   for OBJECT in "${UPDATE_REPO_ARRAY[@]}"; do
+    # Check api limit remainging
+    if [ "${API_LIMIT_REMAINING}" -lt 1 ]; then
```
Comment (bryantson) [inline gh-repo-visibility:683]: could be `-eq 0` Just saying 😄

File: gh-repo-visibility [hunk 2]
```diff
@@ -576,18 +669,21 @@ if [[ -n "${INPUT_FILE_NAME}" ]]; then
...
+      # Check api limit
+      CheckAPILimit
+    fi
     # Get the vars from the string
     ORG_REPO=$(echo "${OBJECT}" | cut -d'|' -f1)
     ORG=$(echo "${ORG_REPO}" | cut -d'/' -f1)
```
Comment (bryantson) [inline gh-repo-visibility:689]: What is `-f1` ?

**Comment by bryantson** [comment]
LGTM :) But you already merged!!

File: gh-repo-visibility (hunk 1 above)
Comment (admiralAwkbar) [reply gh-repo-visibility:683]: yeah good point
lol that was github copiliot... its craxzy

**Pull request merged**